numpy/ma/core.py

```diff
@@ -876,7 +876,8 @@ def __call__(self, a, *args, **kwargs):
                 result = self.f(d, *args, **kwargs)
             # Make a mask
             m = ~umath.isfinite(result)
-            m |= self.domain(d)
+            with np.errstate(divide='ignore', invalid='ignore'):
```

Comment:
Yes, the point of this change was to suppress certain warnings (not errors) that would come up repeatedly. For example in:

    >>> a = np.ma.array([np.nan,1,2], mask=[1,0,0])
    >>> np.log(a)
    RuntimeWarning: invalid value encountered in less_equal
      return umath.less_equal(x, self.critical_value)
    masked_array(data = [-- 0.0 0.6931471805599453],
             mask = [ True False False],
       fill_value = 1e+20)

I think you're right I should move it to the domain definition, and probably the `divide='ignore'` shouldn't be there.